### PR TITLE
/me endpoint

### PR DESCRIPTION
**Pull request by matallo**
re: #12599

### CODE REVIEW

File: app/controllers/carto/api/users_controller.rb [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+        organization_notifications = carto_viewer.received_notifications.unread.map do |n|
+          Carto::Api::ReceivedNotificationPresenter.new(n)
+        end
+
+        render json: {
+          username: current_user.username,
+          user_data: current_user.data,
+          config: frontend_config_hash,
+          upgrade_url: current_user.upgrade_url(request.protocol),
+          isFirstTimeViewingDashboard: !current_user.dashboard_viewed?,
```
Comment (juanignaciosl) [inline app/controllers/carto/api/users_controller.rb]: `isFirstTimeViewingDashboard` and `isJustLoggedIn` should be snake case, unless there's a HUGE problem for frontend to fix this.

File: app/controllers/carto/api/users_controller.rb [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
 
+      def me
+        carto_viewer = Carto::User.find(current_viewer.id)
+        dashboard_notifications = carto_viewer.notifications_for_category(:dashboard)
+        organization_notifications = carto_viewer.received_notifications.unread.map do |n|
+          Carto::Api::ReceivedNotificationPresenter.new(n)
+        end
+
+        render json: {
+          username: current_user.username,
```
Comment (juanignaciosl) [inline app/controllers/carto/api/users_controller.rb]: This method uses both `current_user` and `carto_viewer`,  I think that it should only use `carto_viewer`.

Comment (javitonino) [reply app/controllers/carto/api/users_controller.rb]: I'll add something that I ignored in the first PR.

Most of the fields outside `user_data` can be calculated with info inside `user_data`. We should skip the outside information, meaning:
- Remove `username` (it's directly inside `user_data`)
- Remove `isFirstTimeViewingDashboard` (can be calculated with `last_dashboard_viewed_at` supplied in `user_data`

And probably more. This also implied trivial changes to frontend.

File: app/controllers/carto/api/users_controller.rb (hunk 1 above)
Comment (matallo) [reply app/controllers/carto/api/users_controller.rb]: nooo problem at all

File: app/controllers/carto/api/users_controller.rb (hunk 2 above)
Comment (matallo) [reply app/controllers/carto/api/users_controller.rb]: indeed there are too many keys in the response, I'd rather stick to a minimum

File: app/controllers/carto/api/users_controller.rb (hunk 1 above)
Comment (mayoral) [reply app/controllers/carto/api/users_controller.rb]: Done in https://github.com/CartoDB/cartodb/pull/12675/commits/21037c5164816c0a3b6fb2d3b20b1521691890e4.

@matallo I've done this in a different commit so it's easily revertable if you find any problem changing to snake case in the frontend side.

Comment (matallo) [reply app/controllers/carto/api/users_controller.rb]: i'll manage, thanks mate!

**Comment by matallo** [comment]
totally down with removing duplication, actually I'd just add the minimum required keys that are accessed in https://github.com/CartoDB/cartodb/pull/12679/files#diff-3cc2eaf83cd8934c629f054075d20e87R39

plase make me know if you need me to provide those keys

**Comment by juanignaciosl** [comment]
In addition to addressing duplication, maybe you can take a look at the resulting structure and discuss whether we should keep it "as it" or flatten/aggregate some data. For example, this returns this:

```
...
 user_data: carto_viewer.data,
 config: frontend_config_hash,
...
```

Although it's easy, it feels like being a direct translation of the previous JS variables, which might not be the better solution.

Anyway, this is just a comment and I don't know the frontend needs, so feel free to ignore.

**Comment by matallo** [comment]
for me it's preferible to keep the response at a minimum and if we notice something missing keep adding from there

**Comment by mayoral** [comment]
As commented with @matallo, we are reducing fields exposed in this first version of `/me` endpoint to this list:

```
user_data
default_fallback_basemap
config
dashboard_notifications
is_just_logged_in
is_first_time_viewing_dashboard
```

File: app/controllers/carto/api/users_controller.rb [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
 
       skip_before_filter :api_authorization_required, only: [:get_authenticated_users]
 
       def show
         render json: Carto::Api::UserPresenter.new(uri_user).data
       end
 
+      def me
+        carto_viewer = Carto::User.find(current_viewer.id)
+
```
Comment (amartinj) [inline app/controllers/carto/api/users_controller.rb]: Why are you using a mix of User and Carto::User?

File: app/controllers/carto/api/users_controller.rb [hunk 4]
```diff
@@ -1,15 +1,33 @@
 module Carto
   module Api
     class UsersController < ::Api::ApplicationController
+      include AppAssetsHelper
+      include MapsApiHelper
+      include SqlApiHelper
+      include CartoDB::ConfigUtils
```
Comment (amartinj) [inline app/controllers/carto/api/users_controller.rb]: Why are all those includes here? Don't see where are they being used

Comment (mayoral) [reply app/controllers/carto/api/users_controller.rb]: This was already discussed in #12601.

Long history short, /me needs these helpers to expose some info that we included in some views in the past. No one loves that part, and we hope to get rid of them during the refinements this endpoint will have in the following weeks.

File: app/controllers/carto/api/users_controller.rb (hunk 3 above)
Comment (mayoral) [reply app/controllers/carto/api/users_controller.rb]: Sorry, I didn't saw this earlier. I need Carto::User to expose `notifications_for_category` and `user_frontend_version_or_frontend_version`. The other json attributes can be fetched from User.

Comment (javitonino) [reply app/controllers/carto/api/users_controller.rb]: Yes, but why not use `Carto::User` for everything?

Comment (juanignaciosl) [reply app/controllers/carto/api/users_controller.rb]: Anticipating the answer, `data` is not present at `Carto::User`. Although it's a pity, I'd say that it's not necessary to make this PR more complex with a `*::User` refactor.

Comment (javitonino) [reply app/controllers/carto/api/users_controller.rb]: Well, there is no data, but there is `Carto::UserPresenter.data` which is the same thing. There is even [a test](https://github.com/CartoDB/cartodb/blob/master/spec/models/user_presenter_spec.rb#L36) to ensure they are kept in sync.

**Comment by amartinj** [comment]
Lets move to acceptance then! :)

File: app/controllers/carto/api/users_controller.rb (hunk 3 above)
Comment (juanignaciosl) [reply app/controllers/carto/api/users_controller.rb]: Mixed feelings about this. @mayoral, @javitonino please come to an agreement and move on :).

**Pull request merged**